core/src/main/java/org/elasticsearch/action/support/replication/ReplicatedWriteRequest.java

```diff
@@ -36,6 +38,8 @@
 public abstract class ReplicatedWriteRequest<R extends ReplicatedWriteRequest<R>> extends ReplicationRequest<R> implements WriteRequest<R> {
     private RefreshPolicy refreshPolicy = RefreshPolicy.NONE;
 
+    private long seqNo;
```

Comment:
I just copied over code, but good catch I'll adapt. The problem is that serialization requires a positive number. I'll adapt that too.